packages/eventual-send/src/postponed.js

```diff
@@ -0,0 +1,48 @@
+// @ts-check
+/// <reference types="ses" />
+
+/**
+ * @template T
+ * @typedef {import('.').EHandler<T>} EHandler
+ */
+
+/**
+ * @param {import('.').HandledPromiseConstructor} HandledPromise
+ * @returns {[Required<EHandler<any>>, () => void]}
+ */
+export const makePostponedHandler = HandledPromise => {
+  // Create a simple postponedHandler that just postpones until the
+  // fulfilledHandler is set.
+  let donePostponing;
+  const interlockP = new Promise(resolve => {
+    donePostponing = () => resolve(undefined);
+  });
+
+  const makePostponedOperation = postponedOperation => {
+    // Just wait until the handler is resolved/rejected.
+    return function postpone(x, ...args) {
+      // console.log(`forwarding ${postponedOperation} ${args[0]}`);
+      return new HandledPromise((resolve, reject) => {
+        interlockP
+          .then(_ => {
+            // If targetP is a handled promise, use it, otherwise x.
+            resolve(HandledPromise[postponedOperation](x, ...args));
+          })
+          .catch(reject);
+      });
+    };
+  };
+
+  /** @type {Required<EHandler<any>>} */
```

Comment:
Weird to make all the methods on `EHandler` optional, and then to `Required` them here. I've generally gone the other way, defining something with non-optional methods and then using or defining a `Partial` version of that.

Reply:
I have the types that way because users of the package will generally define only partial handlers.  It's an internal implementation detail that this particular handler must be fully defined.

Reply:
makes sense. resolving.